editor/resources/prelude.lua

```diff
@@ -99,4 +101,282 @@ editor.ui.resource_field = editor.ui.component(function(props)
+        editor.prefs.set(prefs_key .. ".liveupdate", editor.prefs.get("bundle.liveupdate"))
+        editor.prefs.set(prefs_key .. ".contentless", editor.prefs.get("bundle.contentless"))
+    end
+    local config = editor.prefs.get(prefs_key)
+    local errors = errors_fn and errors_fn(config)
+    local show_dialog = requested_dialog or errors
+    if show_dialog then
+        config = editor.ui.show_dialog(dialog_component({config = config, errors = errors}))
+        if not config then
+            error(editor.bundle.abort_message)
+        end
+        editor.prefs.set(prefs_key, config)
+    end
+    return config
+end
+
+function editor.bundle.output_directory(requested_dialog, output_subdir)
+    local output_root = editor.prefs.get("bundle.output-directory")
+    local prompt_overwrite = requested_dialog or #output_root == 0
+    if prompt_overwrite then
+        output_root = editor.ui.show_external_directory_dialog({
+            title = "Output Directory",
+            path = #output_root > 0 and output_root or nil
+        })
+        if output_root then
+            editor.prefs.set("bundle.output-directory", output_root)
+        else
+            error(editor.bundle.abort_message)
+        end
+    end
+    local output_directory = output_root .. "/" .. output_subdir
+    if not prompt_overwrite then return
+        output_directory
+    end
+    local attrs = editor.external_file_attributes(output_directory)
+    if attrs.exists then
+        if attrs.is_file then
+            editor.ui.show_dialog(editor.ui.dialog({
+                title = "Cannot Overwrite File",
+                header = editor.ui.heading({text = "Cannot create a directory"}),
+                content = editor.ui.vertical({ padding = editor.ui.PADDING.LARGE, children = {
+                    editor.ui.paragraph({text = "Cannot create directory at path \"" .. attrs.path .. "\", there is a file with the same name as the directory we are trying to create"})
+                }})
+            }))
+            error(editor.bundle.abort_message)
+        else
+            local overwrite = editor.ui.show_dialog(editor.ui.dialog({
+                title = "Overwrite Existing Directory?",
+                header = editor.ui.vertical({children = {
+                    editor.ui.heading({text = "A directory already exists"}),
+                    editor.ui.paragraph({text = "Overwrite \"" .. attrs.path .. "\"?"})
+                }}),
+                buttons = {
+                    editor.ui.dialog_button({text = "Cancel", cancel = true, result = false}),
+                    editor.ui.dialog_button({text = "Overwrite", default = true, result = true}),
+                }
+            }))
+            if not overwrite then
+                error(editor.bundle.abort_message)
+            end
+        end
+    end
+    return output_directory
+end
+
+function editor.bundle.create(config, output_directory, extra_bob_opts)
```

Comment:
The "invoke bob" step. Mostly corresponds to generic bundle options that are now removed from `editor.pipeline.bob` ns.